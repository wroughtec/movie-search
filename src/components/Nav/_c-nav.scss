@import '../../scss/_mixins/m-container';

.c-nav {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  margin-bottom: 5vh;

  background-color: var(--navBackground);
}

.c-nav__container {
  @include container;
}

.c-nav__links {
  color: var(--navLinks);
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}
