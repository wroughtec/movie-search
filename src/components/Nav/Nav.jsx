import React from 'react';
import { Link } from '@reach/router';
import { home } from 'consts/routes';
import './_c-nav.scss';

export const Nav = ({ resetSearchTerm }) => {
  const handleNav = () => resetSearchTerm('');

  return (
    <div className="c-nav">
      <nav className="c-nav__container">
        <Link to={home} className="c-nav__links" onClick={handleNav}>
          Popular Movies
        </Link>
      </nav>
    </div>
  );
};
