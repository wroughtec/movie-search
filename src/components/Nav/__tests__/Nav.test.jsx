import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from 'components/Nav/Nav';

describe('Nav', () => {
  let component, props;

  beforeEach(() => {
    props = {
      resetSearchTerm: jest.fn()
    };
    component = shallow(<Nav {...props} />);
  });

  test('Renders', () => {
    expect(component).toMatchSnapshot();
  });

  test('function is called', () => {
    component.find('.c-nav__links').simulate('click');
    expect(props.resetSearchTerm).toHaveBeenCalled();
  });
});
