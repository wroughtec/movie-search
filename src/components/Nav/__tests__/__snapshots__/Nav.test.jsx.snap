// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    resetSearchTerm={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="c-nav__container"
      >
        <ForwardRef
          className="c-nav__links"
          onClick={[Function]}
          to="/"
        >
          Popular Movies
        </ForwardRef>
      </nav>,
      "className": "c-nav",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ForwardRef
          className="c-nav__links"
          onClick={[Function]}
          to="/"
        >
          Popular Movies
        </ForwardRef>,
        "className": "c-nav__container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Popular Movies",
          "className": "c-nav__links",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "Popular Movies",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="c-nav__container"
        >
          <ForwardRef
            className="c-nav__links"
            onClick={[Function]}
            to="/"
          >
            Popular Movies
          </ForwardRef>
        </nav>,
        "className": "c-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ForwardRef
            className="c-nav__links"
            onClick={[Function]}
            to="/"
          >
            Popular Movies
          </ForwardRef>,
          "className": "c-nav__container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Popular Movies",
            "className": "c-nav__links",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "Popular Movies",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
